### ISSUE TITLE
callbacks needed for debugger support

### ISSUE THREAD

**Issue by lloyd** [mozilla/jschannel#2]
it would be good to be able to hook a function into jschannel that's invoked immediately before a message is posted, and immediately when one is received.  

Suggested implementation is to change the signature of build to take an object (named arguments)

```
var chan = Channel.build({
    origin: "*",
    onPostMessage: function(origin, msg) { ...},
    onMessageReceived: function(origin, msg) { ...},
    scope: "whatever",
    window: document.getElementById("theIFrame").contentWindow
});
```

**Comment by lloyd** [comment]
16:06 < michaelhanson> lloyd: are you thinking that those onPostMessage and onMessageReceiver methods would be observers, rather than delegates? i.e. the message dispatch would continue normally regardless of what happens in there?
16:06 < lloyd> yeah
16:06 < lloyd> that was my thought.
16:06 < michaelhanson> I would probably rename them to observePostMessage and observeReceiveMessage tehn

**Comment by lloyd** [comment]
build now takes a configuration object. debug output to console is off by default, but may be configured on.  Add observer hooks for before messages are posted and after they're received.  Closed by ebcfeeb3105ec94518a23ddea8e914e93678e8ce.
